### PR TITLE
Support edge cases for DHT key/subkey/value, add tests, update .gitignore for pb2

### PR DESCRIPTION
**Pull request by yhn112**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,3 +75,6 @@ debian/reproducible-experiment-platform
 debian/files
 *.substvars
 *.debhelper.log
+
+# protobuf stuff
+hivemind/proto/*_pb2*
```
Comment (mryab) [inline .gitignore:80]: \n

**Review by justheuristic — APPROVED** [review]

File: tests/test_dht_node.py [hunk 2]
```diff
@@ -433,3 +434,24 @@ async def test_dhtnode_validate(fake_endpoint='127.0.0.721:*'):
     with pytest.raises(ValidationError):
         node2 = await hivemind.DHTNode.create(blacklist_time=999, initial_peers=[f"{LOCALHOST}:{node1.port}"],
                                               endpoint=fake_endpoint)
+
+
+@pytest.mark.forked
+@pytest.mark.asyncio
+async def test_dhtnode_store_get():
```
Comment (justheuristic) [inline tests/test_dht_node.py:441]:
```suggestion
async def test_dhtnode_edge_cases():
```
